contracts/tokenLocking/TokenLocking.sol

```diff
@@ -92,12 +92,21 @@ contract TokenLocking is TokenLockingStorage, DSMath { // ignore-swc-123
     userLocks[_token][msg.sender].lockCount = _lockId;
   }
 
+  // Deprecated interface
   function deposit(address _token, uint256 _amount) public {
-    require(ERC20Extended(_token).transferFrom(msg.sender, address(this), _amount), "colony-token-locking-transfer-failed"); // ignore-swc-123
+    deposit(_token, _amount, false);
+  }
 
-    makeConditionalDeposit(_token, _amount, msg.sender);
+  function deposit(address _token, uint256 _amount, bool _force) public tokenNotLocked(_token, _force) {
+    require(ERC20Extended(_token).transferFrom(msg.sender, address(this), _amount), "colony-token-locking-transfer-failed"); // ignore-swc-123
 
     Lock storage lock = userLocks[_token][msg.sender];
+    lock.balance = add(lock.balance, _amount);
+
+    // Handle the deprecated pendingBalance, if any (idempotent operation)
+    lock.balance = add(lock.balance, lock.DEPRECATED_pendingBalance);
+    delete lock.DEPRECATED_pendingBalance;
```

Comment:
This should only happen if `_force` is true, I think? But otherwise, yeah, this is a nice solution.

Reply:
I'll go with that. An alternative would be to check if the tokens is locked, but that's more gas for a conditional which will only be flipped once.

Reply:
Oh, you're right, that is a better trigger. But gassier, as you say. I agree it's not worth it.

Reply:
This still needs to be inside a conditional.